src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java

```diff
@@ -0,0 +1,232 @@
+    "The NotifyForFields values are:\n" +
+    "All - Notifications are generated for all record field changes, provided the evaluated records match " +
+    "the criteria specified in the WHERE clause.\n" +
+    "Referenced (default) - Changes to fields referenced in the SELECT and WHERE clauses are evaluated. " +
+    "Notifications are generated for the evaluated " +
+    "records only if they match the criteria specified in the WHERE clause.\n" +
+    "Select - Changes to fields referenced in the SELECT clause are evaluated. Notifications are generated " +
+    "for the evaluated records only if they match the criteria specified in the WHERE clause.\n" +
+    "Where - Changes to fields referenced in the WHERE clause are evaluated. Notifications are generated " +
+    "for the evaluated records only if they match the criteria specified in the WHERE clause.")
+  @Nullable
+  @Name("pushTopicNotifyForFields")
+  private String pushTopicNotifyForFields;
+
+  public SalesforceStreamingSourceConfig(String referenceName, String clientId, String clientSecret,
+                                         String username, String password, String loginUrl,
+                                         String errorHandling, String pushTopicName) {
+    super(referenceName, clientId, clientSecret, username, password, loginUrl, errorHandling);
+    this.pushTopicName = pushTopicName;
+  }
+
+  public String getPushTopicName() {
+    return pushTopicName;
+  }
+
+  public String getPushTopicQuery() {
+    return pushTopicQuery;
+  }
+
+  public Boolean isPushTopicNotifyCreate() {
+    return pushTopicNotifyCreate.equals(ENABLED_KEYWORD);
+  }
+
+  public Boolean isPushTopicNotifyUpdate() {
+    return pushTopicNotifyUpdate.equals(ENABLED_KEYWORD);
+  }
+
+  public Boolean isPushTopicNotifyDelete() {
+    return pushTopicNotifyDelete.equals(ENABLED_KEYWORD);
+  }
+
+  public String getPushTopicNotifyForFields() {
+    return pushTopicNotifyForFields;
+  }
+
+  /**
+   * Asserts that pushTopic on Salesforce server has the same parameters as specified in config.
+   * If they are different {@link java.lang.IllegalArgumentException} is thrown.
+   *
+   * If pushTopic does not exist it is created.
+   */
+  public void ensurePushTopicExistAndWithCorrectFields() {
+    if (containsMacro(PROPERTY_PUSHTOPIC_NAME) || containsMacro(PROPERTY_PUSHTOPIC_QUERY)) {
+      return;
+    }
+
+    try {
+      PartnerConnection partnerConnection = new PartnerConnection(
+        Authenticator.createConnectorConfig(this.getAuthenticatorCredentials()));
+
+      SObject pushTopic = fetchPushTopicByName(partnerConnection, pushTopicName);
+
+      if (pushTopic == null) {
+        LOG.info("Creating PushTopic " + pushTopicName);
+
+        if (pushTopicQuery == null || pushTopicQuery.isEmpty()) {
+          throw new IllegalArgumentException("Please specify pushTopicQuery. PushTopic creation failed");
```

Comment:
validation methods should throw only InvalidStageException or InvalidConfigPropertyException exceptions